### PR TITLE
fix dense tensor miss copy when pass start. fix nan problem in wasq.

### PR DESCRIPTION
**Pull request by tiancaitzp**
### PR types

Bug fixes

### PR changes

APIs

### Describe

fix dense tensor miss copy when pass start. fix nan problem in wasq.

### CODE REVIEW

File: paddle/fluid/operators/pull_box_extended_sparse_op.h [hunk 1]
```diff
@@ -159,6 +162,9 @@ static void PullBoxExtendedSparseFunctor(
         }
         auto *output_extend = outputs_extend[expand_offset]->mutable_data<T>(ctx.GetPlace());
         all_values[i + slot_size] = reinterpret_cast<float*>(output_extend);
+        if(outputs_extend[expand_offset]->numel()==0) {
```
Comment (jack603047588) [inline paddle/fluid/operators/pull_box_extended_sparse_op.h]: 等号前后的空格

File: paddle/fluid/operators/pull_box_extended_sparse_op.h [hunk 2]
```diff
@@ -145,6 +145,9 @@ static void PullBoxExtendedSparseFunctor(
         }
         auto *output = outputs[embedx_offset]->mutable_data<T>(ctx.GetPlace());
         all_values[i] = reinterpret_cast<float*>(output);
+        if (outputs[embedx_offset]->numel() == 0) {
+            all_values[i] = nullptr;
```
Comment (jack603047588) [inline paddle/fluid/operators/pull_box_extended_sparse_op.h]: 2格缩进

File: paddle/fluid/operators/pull_box_extended_sparse_op.h [hunk 3]
```diff
@@ -159,6 +162,9 @@ static void PullBoxExtendedSparseFunctor(
         }
         auto *output_extend = outputs_extend[expand_offset]->mutable_data<T>(ctx.GetPlace());
         all_values[i + slot_size] = reinterpret_cast<float*>(output_extend);
+        if(outputs_extend[expand_offset]->numel()==0) {
+            all_values[i] = nullptr;
```
Comment (jack603047588) [inline paddle/fluid/operators/pull_box_extended_sparse_op.h]: 2格缩进

File: paddle/fluid/operators/pull_box_extended_sparse_op.h (hunk 2 above)
Comment (tiancaitzp) [reply paddle/fluid/operators/pull_box_extended_sparse_op.h]: done

File: paddle/fluid/operators/pull_box_extended_sparse_op.h (hunk 1 above)
Comment (tiancaitzp) [reply paddle/fluid/operators/pull_box_extended_sparse_op.h]: done

File: paddle/fluid/operators/pull_box_extended_sparse_op.h (hunk 3 above)
Comment (tiancaitzp) [reply paddle/fluid/operators/pull_box_extended_sparse_op.h]: done

**Review by jack603047588 — APPROVED** [review]

**Pull request merged**